examples/index.md

```diff
@@ -162,3 +162,292 @@ Content-Type: application/vnd.api+json
   }
 }
 ```
+
+## Error Objects
+
+Examples of how [error objects](http://jsonapi.org/format/#error-objects) work.
+
+### Responses to failed resource creation:
+
+Response with one error object, with attribute error:
+
+```http
+HTTP/1.1 422 Unprocessable Entity
+Content-Type: application/vnd.api+json
+
+{
+  "errors": [
+    {
+      "title":  "Invalid Attribute",
+      "source": { "pointer": "data/attributes/first-name" },
+      "detail": "First name must contain at least three characters."
+    }
+  ]
+}
+```
+
+Notice that the attribute `first-name` is dasherized.  Dasherized attributes are preferred
+over underscored.
+
+Notice also that the "detail" is specific to this occurrence of the problem,
+whereas the "title" is more generic.
+
+Notice that the "source" points to the associated attribute with the error.
+
+Response with one error object, with relationship error:
+
+```http
+HTTP/1.1 422 Unprocessable Entity
+Content-Type: application/vnd.api+json
+
+{
+  "errors": [
+    {
+      "title":  "Missing Association",
+      "source": { "pointer": "data/relationships/author" },
+      "detail": "Post author cannot be blank."
+    }
+  ]
+}
+```
+
+Notice that the "source" points to the associated relationship with the error.
+
+Response with one error, using a "code":
+
+```http
+HTTP/1.1 422 Unprocessable Entity
+Content-Type: application/vnd.api+json
+
+{
+  "errors": [
+    {
+      "title": "Adapter Error",
+      "code": "ds.errors.invalid-error-expects-json-api-format",
+      "detail": "The adapter rejected the commit because it was invalid"
+    }
+  ]
+}
+```
+
+Response with status:
+
+```http
+HTTP/1.1 400 Bad Request
+Content-Type: application/vnd.api+json
+
+{
+  "errors": [
+    {
+      "status": "403",
+      "source": { "pointer": "data/attributes/secret-powers" },
+      "detail": "Access to secret powers not authorized on Sundays."
+    },
+    {
+      "status": "404",
+      "source": { "pointer": "data/attributes/vorpal-blade" },
+      "detail": "'Vorpal Blade' is not an attribute of mimsy borogroves."
+    },
+    {
+      "status": "422",
+      "source": { "pointer": "data/attributes/volume" },
+      "detail": "Volume does not, in fact, go to 11."
+    },
+    {
+      "status": "500",
+      "title": "The backend responded with an error",
+      "detail": "Reputation service not responding after three requests."
+    },
+    {
+      "status": "400",
+      "detail": "JSON parse error - Expecting property name enclosed in double quotes: line 1 column 2 (char 1)"
+    }
+  ]
+}
+```
+
+The only uniqueness constraint on error objects in a response is the `id` field.
+Multiple errors on the same attribute may be described by an error object
+for each error.
+
+Multiple errors on `first-name` attribute:
+
+```http
+HTTP/1.1 422 Unprocessable Entity
```

Comment:
422 works here, but 400 is [also valid](http://stackoverflow.com/a/20215807/1261879), and some people still don't like using status codes from WebDAV for non-webDAV apps.

Therefore, maybe you could add a line at the bottom of this example saying that a 400 response would also be ok, and linking to that SO answer? I don't want JSON API to appear to be taking a position on 400 vs. 422 by only using one in our examples.